tests/test_report_manager.py

```diff
@@ -55,98 +87,112 @@ def setUp(self):
+        self._setUp_managers("test_model1,test_model2")
+        result_comparator = ResultComparator(
+            metric_objectives={"perf_throughput": 10})
 
-        avg_gpu_metrics = {0: {'gpu_used_memory': 6000, 'gpu_utilization': 60}}
+        avg_gpu_metrics = {0: [GPUUsedMemory(6000), GPUUtilization(60)]}
 
         for i in range(10):
-            avg_non_gpu_metrics = {
-                'perf_throughput': 100 + 10 * i,
-                'perf_latency': 4000
-            }
-            self.report_manager.add_result(
-                report_key='test_report1',
-                result=construct_result(
-                    avg_gpu_metric_values=avg_gpu_metrics,
-                    avg_non_gpu_metric_values=avg_non_gpu_metrics,
-                    comparator=self.result_comparator))
+            self.model_config["name"] = f"test_model1_report_{i}"
+            model_config = ModelConfig.create_from_dictionary(self.model_config)
+            run_config = RunConfig("test_model1", model_config,
+                                   self.perf_config)
+
+            avg_non_gpu_metrics = [
+                PerfThroughput(100 + 10 * i),
+                PerfLatency(4000),
+                CPUUsedRAM(1000)
+            ]
+            measurement = Measurement(gpu_data=avg_gpu_metrics,
+                                      non_gpu_data=avg_non_gpu_metrics,
+                                      perf_config=self.perf_config)
+            measurement.set_result_comparator(result_comparator)
+
+            self.result_manager.add_measurement(run_config, measurement)
 
         for i in range(5):
-            avg_non_gpu_metrics = {
-                'perf_throughput': 200 + 10 * i,
-                'perf_latency': 4000
-            }
-            self.report_manager.add_result(
-                report_key='test_report2',
-                result=construct_result(
-                    avg_gpu_metric_values=avg_gpu_metrics,
-                    avg_non_gpu_metric_values=avg_non_gpu_metrics,
-                    comparator=self.result_comparator))
+            self.model_config["name"] = f"test_model2_report_{i}"
+            model_config = ModelConfig.create_from_dictionary(self.model_config)
+            run_config = RunConfig("test_model2", model_config,
+                                   self.perf_config)
+
+            avg_non_gpu_metrics = [
+                PerfThroughput(200 + 10 * i),
+                PerfLatency(4000),
+                CPUUsedRAM(1000)
+            ]
+            measurement = Measurement(gpu_data=avg_gpu_metrics,
+                                      non_gpu_data=avg_non_gpu_metrics,
+                                      perf_config=self.perf_config)
+            measurement.set_result_comparator(result_comparator)
+
+            self.result_manager.add_measurement(run_config, measurement)
+
+        self.result_manager.compile_and_sort_results()
+        with unittest.mock.patch(
```

Comment:
Look into using the `mock_io.py` classes. You can instantiate the io_mock in the setUp function, just like we do with all the other mocks. Look at `test_state_manager.py` to see if you can use that here as well.

Reply:
Updated